src/uct/tcp/tcp_cm.c

```diff
@@ -529,6 +582,10 @@ static ucs_status_t uct_tcp_cm_conn_complete(uct_tcp_ep_t *ep,
         goto out;
     }
 
+    /* reset connection retries counter for futher re-use on the next
+     * connection establishmnet iterations if it fails by some reason */
+    ep->conn_retries = 0;
```

Comment:
what is the next connection establishment iteration? if ep is connected why would it try to connect again?

Reply:
the remote side should close the "losing" socket only after receiving connection request on the new socket.
and if we gent any error after successfully sending conn_req, don't do any retries, invoke err handler immediately. we send retries only on conn_req.

Reply:
done